pkg/controller/signup_test.go

```diff
@@ -222,147 +235,150 @@ func (s *TestSignupSuite) TestSignupGetHandler() {
+		require.NotEmpty(s.T(), updatedUserSignup.Annotations[crtapi.UserVerificationExpiryAnnotationKey])
+		require.Equal(s.T(), "2", updatedUserSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey])
+		require.Equal(s.T(), "fd276563a8232d16620da8ec85d0575f", updatedUserSignup.Labels[crtapi.UserSignupUserPhoneHashLabelKey])
 	})
 
 	s.Run("init verification success phone number with dashes", func() {
+		gock.New("https://api.twilio.com").
+			Reply(http.StatusNoContent).
+			BodyString("")
+
 		data := []byte(`{"phone_number": "226-821-3044", "country_code": "1"}`)
 		rr := setup(s.T(), handler, gin.Param{}, data, userID, http.MethodPut, "/api/v1/signup/verification")
+		require.Equal(s.T(), http.StatusNoContent, rr.Code)
+
+		updatedUserSignup, err := s.FakeUserSignupClient.Get(userSignup.Name)
+		require.NoError(s.T(), err)
 
-		assertVerification(s.T(), storedUserSignup, http.StatusNoContent, rr.Code, userID, "12268213044",
-			"jsmith@redhat.com", "1", storedVerificationCode, verificationInitTimeStamp)
+		require.NotEmpty(s.T(), updatedUserSignup.Annotations[crtapi.UserSignupVerificationCodeAnnotationKey])
+		require.NotEmpty(s.T(), updatedUserSignup.Annotations[crtapi.UserSignupVerificationInitTimestampAnnotationKey])
+		require.NotEmpty(s.T(), updatedUserSignup.Annotations[crtapi.UserVerificationExpiryAnnotationKey])
+		require.Equal(s.T(), "3", updatedUserSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey])
+		require.Equal(s.T(), "fd276563a8232d16620da8ec85d0575f", updatedUserSignup.Labels[crtapi.UserSignupUserPhoneHashLabelKey])
 	})
 	s.Run("init verification success phone number with spaces", func() {
+		gock.New("https://api.twilio.com").
+			Reply(http.StatusNoContent).
+			BodyString("")
+
 		data := []byte(`{"phone_number": "2 2 6 8 2 1 3 0 4 4", "country_code": "1"}`)
 		rr := setup(s.T(), handler, gin.Param{}, data, userID, http.MethodPut, "/api/v1/signup/verification")
+		require.Equal(s.T(), http.StatusNoContent, rr.Code)
 
-		// Check the status code is what we expect.
-		assert.Equal(s.T(), http.StatusNoContent, rr.Code, "handler returned wrong status code")
+		updatedUserSignup, err := s.FakeUserSignupClient.Get(userSignup.Name)
+		require.NoError(s.T(), err)
 
-		assertVerification(s.T(), storedUserSignup, http.StatusNoContent, rr.Code, userID, "12268213044",
-			"jsmith@redhat.com", "1", storedVerificationCode, verificationInitTimeStamp)
+		require.NotEmpty(s.T(), updatedUserSignup.Annotations[crtapi.UserSignupVerificationCodeAnnotationKey])
+		require.NotEmpty(s.T(), updatedUserSignup.Annotations[crtapi.UserSignupVerificationInitTimestampAnnotationKey])
+		require.NotEmpty(s.T(), updatedUserSignup.Annotations[crtapi.UserVerificationExpiryAnnotationKey])
+		require.Equal(s.T(), "4", updatedUserSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey])
+		require.Equal(s.T(), "fd276563a8232d16620da8ec85d0575f", updatedUserSignup.Labels[crtapi.UserSignupUserPhoneHashLabelKey])
```

Comment:
The four tests above are almost identical, right? The only difference is the phone format. Can we use some custom assertion for that instead of duplicating the code?

Something like:
```
assertInitVerificationSuccess := func(phone string) {
    gock.New(....)
    ...
    setup()...
    ...
    require....
}

assertInitVerificationSuccess("2 2 6 ...")
assertInitVerificationSuccess("2-26 ...")
....
```

Aslo, I would rename `setup()` to something like `initVerification()` because this function is not only about setting up some test env. It actually does the action we want to test.

Reply:
Good suggestions, implemented.